tests/suites/test_suite_psa_crypto_metadata.function

```diff
@@ -735,3 +735,143 @@ void lifetime(int lifetime_arg, int classification_flags,
     TEST_EQUAL(PSA_KEY_LIFETIME_GET_LOCATION(lifetime), location);
 }
 /* END_CASE */
+
+/* BEGIN_CASE */
+void constant_sizes()
+{
+    /* The goal of this test function is to verify that sizez macros
+     * are compile-time constants. If they aren't, the build will fail
+     * if the compiler doesn't accept variable-length arrays. */
```

Comment:
Does this mean the test will always pass if the compiler does accept VLAs? We could make this more robust by also testing using `__builtin_constant_p()` where available (see https://gcc.gnu.org/onlinedocs/gcc-3.4.0/gcc/Other-Builtins.html#:~:text=int%20__builtin_constant_p%20(exp) - also supported by clang and MSVC)

Reply:
The code will compile if the compiler accepts VLAs, yes. But so what? As long as we do at least one build in our CI that rejects VLAs and has the right options to build this code snippet, this code serves its purpose. And we do, comfortably: all of our CMake builds with Clang or GCC pass `-Wvla -Werror`.

As I understand `__builtin_constant_p`, it could give false negatives if we accidentally use an inline function in the definition of a macro and the function gets inlined: we'd have to rely on catching a failure in unoptimized builds. Which we do, but not nearly as much as guaranteed-no-VLA builds.

Reply:
OK, if we are passing `-Wvla` then there's no risk.